Adding support for custom protocols

giovannipizzi opened the pull request:
Very often one wants a custom protocol, but it's not convenient to add a protocol definition in this repo for every independent project. Therefore, I'm here supporting the possibility to pass the protocol as a dictionary.

Note: For now, only implemented for QE and abinit, needs to be implemented for all if we agree on how I did it.

Note: In this PR I also fixing a couple of bugs in the ACWF relax for ABINIT.

giovannipizzi committed:
Adding support for custom protocols

For now, only implemented for QE and abinit, needs to be implemented for all

Also, fixing a couple of bugs in the ACWF relax for ABINIT

azadoks commented:
Thanks, Giovanni!
This would have been a huge time saver when working on the pseudo improvements for the verification work; hopefully it will be one for future efforts.

I'm not sure when I renamed the forces output in the plugin (it's even marked for backwards-compatibility), but great that you caught it.

I'll give this a deeper look on Monday.

azadoks commented:
After a second look, this looks reasonable to me and shouldn't be too hard to add for all the codes 👍🏼

giovannipizzi committed:
Adding possibility to specify cutoffs in the protocol

This allows to override the protocol ones by the launching user
by using a custom protocol

giovannipizzi committed:
Fixing pre-commit issues

azadoks reviewed (commented):
Looks good!

giovannipizzi committed:
Adding support for common protocols in VASP

giovannipizzi committed:
Fixing default protocol

giovannipizzi committed:
Fix the type of a port

giovannipizzi committed:
Merge branch 'master' of github.com:aiidateam/aiida-common-workflows into add_custom_protocols

giovannipizzi committed:
Adding support for explicit k-points in VASP generator

ahkole committed:
Draft support for non-collinear spins for the CommonRelaxInputGenerator

ahkole committed:
Add non-collinear spin to siesta relax workchain

ahkole committed:
Correctly set spin parameter for siesta relax workchain for non-collinear spins

ahkole committed:
Move to_spherical to separate util module + improve robustness of spin_type handling in siesta relax workchain

ahkole committed:
Clarifying documentation for 'magnetization_per_site'

giovannipizzi committed:
Merge branch 'master' of github.com:aiidateam/aiida-common-workflows into add_custom_protocols

ahkole committed:
Refactor validation of magnetization_per_site

ahkole committed:
Refactor setting magnetization in siesta relax input generator

ahkole committed:
Fix punctuation in docs

ahkole committed:
Add check for magnetization_per_site that line contains correct number of entries

giovannipizzi committed:
Merge pull request #2 from ahkole/relax-non-collinear-spin

Add support for non-collinear spin in base relax workflow

t-reents committed:
Merge branch 'master' into add_custom_protocols

mariaandolfatto committed:
add noncollinear case for qe (#4)

* add noncollinear case for qe

---------

Co-authored-by: Timo Reents <timo.reents1@gmail.com>

ahkole committed:
Add custom protocols siesta (#1)

* Implement custom_protocol for siesta

* Support directly supplying KpointsData node for siesta engine

* Apply pre-commit fixes

* Convert AiiDA Dicts in SIESTA custom_protocol to plain dicts

* Allow setting of lua inputs in custom protocol for siesta
